### ISSUE TITLE
Cant connect to MCU after update 

### ISSUE THREAD

**Issue by StefanMeGit** [Klipper3d/klipper#5012]
Hi Klipper Friends,

i updated my system since a while and run into some major issues. At first i had a hard time to flash the ARDUINO DUE. The flash process stopped randomly. I figured out, that Octoprint is continously try to reconnect to the DUE and interrupt the flash process. I stopped the octoprint service... flashing worked like a charm.
![image](https://user-images.githubusercontent.com/16327026/145368952-b44942dc-9c5d-4385-9e0d-80c291921d71.png)

After restaring the RPI, i still cannt connect to the MCU. Any ideas what i am doing wrong? The Serial is the same as the flashing serial.
![image](https://user-images.githubusercontent.com/16327026/145370602-9d18f546-5f7c-44a9-9ef0-ccc054c3f17a.png)

The system worked before the update just fine.
[klippy.log](https://github.com/Klipper3d/klipper/files/7683804/klippy.log)

**Comment by Sineos** [comment]
Verify if your serial path
```
[mcu]
serial = /dev/serial/by-id/usb-Arduino__www.arduino.cc__Arduino_Due_Prog._Port_55839313638351F041C2-if00
```
is still correct.

If yes:
1. Repeat the flash process
2. Make sure to power down the board completely
3. Reattach board to the Pi / power up
4. Post output of `ls /dev/serial/by-id/*` command
4. Post output of `dmesg` command
5. Post output of `lsusb` command

**Comment by StefanMeGit** [comment]
Thanks for the help. 

just did everything u mentioned exepct of power off an on the arduino (just not at  home at the moment... did everything by remote).  I ll do it as soon as i get home. 

The serial looks correct.
![image](https://user-images.githubusercontent.com/16327026/145385218-1f6ab53e-2713-4fcf-b283-80ae219ccdf2.png)

```
pi@octopi:~ $ dmesg
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.10.63-v7+ (dom@buildbot) (arm-linux-gnueabihf-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1496 SMP Wed Dec 1 15:58:11 GMT 2021
[    0.000000] CPU: ARMv7 Processor [410fd034] revision 4 (ARMv7), cr=10c5383d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Raspberry Pi 3 Model B Plus Rev 1.3
[    0.000000] random: fast init done
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Reserved memory: created CMA memory pool at 0x34000000, size 64 MiB
[    0.000000] OF: reserved mem: initialized node linux,cma, compatible id shared-dma-pool
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000000000-0x0000000037ffffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000037ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000037ffffff]
[    0.000000] On node 0 totalpages: 229376
[    0.000000]   DMA zone: 2016 pages used for memmap
[    0.000000]   DMA zone: 0 pages reserved
[    0.000000]   DMA zone: 229376 pages, LIFO batch:63
[    0.000000] percpu: Embedded 20 pages/cpu s50828 r8192 d22900 u81920
[    0.000000] pcpu-alloc: s50828 r8192 d22900 u81920 alloc=20*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 227360
[    0.000000] Kernel command line: coherent_pool=1M 8250.nr_uarts=0 snd_bcm2835.enable_compat_alsa=0 snd_bcm2835.enable_hdmi=1 bcm2708_fb.fbwidth=656 bcm2708_fb.fbheight=512 bcm2708_fb.fbswap=1 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  console=ttyS0,115200 console=tty1 root=PARTUUID=874acb7a-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait
[    0.000000] Kernel parameter elevator= does not have any effect anymore.
               Please use sysfs to set IO scheduler for individual devices.
[    0.000000] Dentry cache hash table entries: 131072 (order: 7, 524288 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 826028K/917504K available (10240K kernel code, 1311K rwdata, 2948K rodata, 1024K init, 859K bss, 25940K reserved, 65536K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] ftrace: allocating 32054 entries in 63 pages
[    0.000000] ftrace: allocated 63 pages with 6 groups
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Rude variant of Tasks RCU enabled.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x3ac/0x580 with crng_init=1
[    0.000000] arch_timer: cp15 timer(s) running at 19.20MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x46d987e47, max_idle_ns: 440795202767 ns
[    0.000007] sched_clock: 56 bits at 19MHz, resolution 52ns, wraps every 4398046511078ns
[    0.000025] Switching to timer-based delay loop, resolution 52ns
[    0.000315] Console: colour dummy device 80x30
[    0.001174] printk: console [tty1] enabled
[    0.001241] Calibrating delay loop (skipped), value calculated using timer frequency.. 38.40 BogoMIPS (lpj=192000)
[    0.001299] pid_max: default: 32768 minimum: 301
[    0.001522] LSM: Security Framework initializing
[    0.001780] Mount-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.001829] Mountpoint-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.003308] cgroup: Disabling memory control group subsystem
[    0.003596] CPU: Testing write buffer coherency: ok
[    0.004114] CPU0: thread -1, cpu 0, socket 0, mpidr 80000000
[    0.005374] Setting up static identity map for 0x100000 - 0x10003c
[    0.005583] rcu: Hierarchical SRCU implementation.
[    0.006528] smp: Bringing up secondary CPUs ...
[    0.007706] CPU1: thread -1, cpu 1, socket 0, mpidr 80000001
[    0.008989] CPU2: thread -1, cpu 2, socket 0, mpidr 80000002
[    0.010307] CPU3: thread -1, cpu 3, socket 0, mpidr 80000003
[    0.010463] smp: Brought up 1 node, 4 CPUs
[    0.010517] SMP: Total of 4 processors activated (153.60 BogoMIPS).
[    0.010548] CPU: All CPU(s) started in HYP mode.
[    0.010575] CPU: Virtualization extensions available.
[    0.011579] devtmpfs: initialized
[    0.029355] VFP support v0.3: implementor 41 architecture 3 part 40 variant 3 rev 4
[    0.029652] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.029712] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.032974] pinctrl core: initialized pinctrl subsystem
[    0.034183] NET: Registered protocol family 16
[    0.038478] DMA: preallocated 1024 KiB pool for atomic coherent allocations
[    0.044230] audit: initializing netlink subsys (disabled)
[    0.045146] thermal_sys: Registered thermal governor 'step_wise'
[    0.046004] audit: type=2000 audit(0.040:1): state=initialized audit_enabled=0 res=1
[    0.046233] hw-breakpoint: found 5 (+1 reserved) breakpoint and 4 watchpoint registers.
[    0.046274] hw-breakpoint: maximum watchpoint size is 8 bytes.
[    0.046581] Serial: AMBA PL011 UART driver
[    0.064879] bcm2835-mbox 3f00b880.mailbox: mailbox enabled
[    0.080145] raspberrypi-firmware soc:firmware: Attached to firmware from 2021-12-01T15:08:00, variant start_x
[    0.090157] raspberrypi-firmware soc:firmware: Firmware hash is 71bd3109023a0c8575585ba87cbb374d2eeb038f
[    0.134342] Kprobes globally optimized
[    0.139308] bcm2835-dma 3f007000.dma: DMA legacy API manager, dmachans=0x1
[    0.141675] SCSI subsystem initialized
[    0.141958] usbcore: registered new interface driver usbfs
[    0.142044] usbcore: registered new interface driver hub
[    0.142137] usbcore: registered new device driver usb
[    0.144123] clocksource: Switched to clocksource arch_sys_counter
[    1.917713] VFS: Disk quotas dquot_6.6.0
[    1.917856] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[    1.918082] FS-Cache: Loaded
[    1.918337] CacheFiles: Loaded
[    1.928415] NET: Registered protocol family 2
[    1.928710] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    1.930154] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    1.930242] TCP established hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    1.930490] TCP bind hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    1.930705] TCP: Hash tables configured (established 8192 bind 8192)
[    1.930906] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    1.930984] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    1.931693] NET: Registered protocol family 1
[    1.932547] RPC: Registered named UNIX socket transport module.
[    1.932581] RPC: Registered udp transport module.
[    1.932610] RPC: Registered tcp transport module.
[    1.932639] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    1.934521] hw perfevents: enabled with armv7_cortex_a7 PMU driver, 7 counters available
[    1.938343] Initialise system trusted keyrings
[    1.938658] workingset: timestamp_bits=14 max_order=18 bucket_order=4
[    1.948266] zbud: loaded
[    1.950361] FS-Cache: Netfs 'nfs' registered for caching
[    1.951288] NFS: Registering the id_resolver key type
[    1.951348] Key type id_resolver registered
[    1.951378] Key type id_legacy registered
[    1.951544] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    1.951580] nfs4flexfilelayout_init: NFSv4 Flexfile Layout Driver Registering...
[    1.952808] Key type asymmetric registered
[    1.952842] Asymmetric key parser 'x509' registered
[    1.952914] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 249)
[    1.952954] io scheduler mq-deadline registered
[    1.952984] io scheduler kyber registered
[    1.956641] bcm2708_fb soc:fb: FB found 1 display(s)
[    1.971211] Console: switching to colour frame buffer device 82x32
[    1.977160] bcm2708_fb soc:fb: Registered framebuffer for display 0, size 656x512
[    1.986125] bcm2835-rng 3f104000.rng: hwrng registered
[    1.989692] vc-mem: phys_addr:0x00000000 mem_base=0x3ec00000 mem_size:0x40000000(1024 MiB)
[    1.997032] gpiomem-bcm2835 3f200000.gpiomem: Initialised: Registers at 0x3f200000
[    2.016099] brd: module loaded
[    2.031648] loop: module loaded
[    2.036769] Loading iSCSI transport class v2.0-870.
[    2.041642] libphy: Fixed MDIO Bus: probed
[    2.045280] usbcore: registered new interface driver lan78xx
[    2.048603] usbcore: registered new interface driver smsc95xx
[    2.051810] dwc_otg: version 3.00a 10-AUG-2012 (platform bus)
[    2.783340] Core Release: 2.80a
[    2.786609] Setting default values for core params
[    2.789847] Finished setting default values for core params
[    2.993440] Using Buffer DMA mode
[    2.996565] Periodic Transfer Interrupt Enhancement - disabled
[    2.999757] Multiprocessor Interrupt Enhancement - disabled
[    3.002967] OTG VER PARAM: 0, OTG VER FLAG: 0
[    3.006149] Dedicated Tx FIFOs mode

[    3.009758] WARN::dwc_otg_hcd_init:1074: FIQ DMA bounce buffers: virt = b4114000 dma = 0xf4114000 len=9024
[    3.019066] FIQ FSM acceleration enabled for :
               Non-periodic Split Transactions
               Periodic Split Transactions
               High-Speed Isochronous Endpoints
               Interrupt/Control Split Transaction hack enabled
[    3.034175] dwc_otg: Microframe scheduler enabled

[    3.034256] WARN::hcd_init_fiq:457: FIQ on core 1

[    3.039828] WARN::hcd_init_fiq:458: FIQ ASM at 807c9984 length 36

[    3.045229] WARN::hcd_init_fiq:497: MPHI regs_base at b8810000
[    3.050818] dwc_otg 3f980000.usb: DWC OTG Controller
[    3.053713] dwc_otg 3f980000.usb: new USB bus registered, assigned bus number 1
[    3.056687] dwc_otg 3f980000.usb: irq 89, io mem 0x00000000
[    3.059674] Init: Port Power? op_state=1
[    3.062571] Init: Power Port (0)
[    3.065775] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.10
[    3.071624] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.074753] usb usb1: Product: DWC OTG Controller
[    3.077799] usb usb1: Manufacturer: Linux 5.10.63-v7+ dwc_otg_hcd
[    3.080863] usb usb1: SerialNumber: 3f980000.usb
[    3.084683] hub 1-0:1.0: USB hub found
[    3.087733] hub 1-0:1.0: 1 port detected
[    3.091954] dwc_otg: FIQ enabled
[    3.091968] dwc_otg: NAK holdoff enabled
[    3.091979] dwc_otg: FIQ split-transaction FSM enabled
[    3.091998] Module dwc_common_port init
[    3.092315] usbcore: registered new interface driver usb-storage
[    3.095639] mousedev: PS/2 mouse device common for all mice
[    3.099797] bcm2835-wdt bcm2835-wdt: Broadcom BCM2835 watchdog timer
[    3.105859] sdhci: Secure Digital Host Controller Interface driver
[    3.108920] sdhci: Copyright(c) Pierre Ossman
[    3.112581] mmc-bcm2835 3f300000.mmcnr: could not get clk, deferring probe
[    3.116498] sdhost-bcm2835 3f202000.mmc: could not get clk, deferring probe
[    3.119953] sdhci-pltfm: SDHCI platform and OF driver helper
[    3.125062] ledtrig-cpu: registered to indicate activity on CPUs
[    3.128736] hid: raw HID events driver (C) Jiri Kosina
[    3.132163] usbcore: registered new interface driver usbhid
[    3.135475] usbhid: USB HID core driver
[    3.143890] Initializing XFRM netlink socket
[    3.147188] NET: Registered protocol family 17
[    3.150514] Key type dns_resolver registered
[    3.154306] Registering SWP/SWPB emulation handler
[    3.157596] registered taskstats version 1
[    3.160718] Loading compiled-in X.509 certificates
[    3.164781] Key type ._fscrypt registered
[    3.167862] Key type .fscrypt registered
[    3.170819] Key type fscrypt-provisioning registered
[    3.185551] uart-pl011 3f201000.serial: cts_event_workaround enabled
[    3.188620] 3f201000.serial: ttyAMA0 at MMIO 0x3f201000 (irq = 114, base_baud = 0) is a PL011 rev2
[    3.197017] bcm2835-power bcm2835-power: Broadcom BCM2835 power domains driver
[    3.201866] mmc-bcm2835 3f300000.mmcnr: mmc_debug:0 mmc_debug2:0
[    3.205171] mmc-bcm2835 3f300000.mmcnr: DMA channel allocated
[    3.214280] Indeed it is in host mode hprt0 = 00021501
[    3.291894] sdhost: log_buf @ (ptrval) (f4113000)
[    3.332227] mmc1: queuing unknown CIS tuple 0x80 (2 bytes)
[    3.337065] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    3.341770] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    3.347674] mmc1: queuing unknown CIS tuple 0x80 (7 bytes)
[    3.350497] mmc0: sdhost-bcm2835 loaded - DMA enabled (>1)
[    3.356277] of_cfs_init
[    3.359050] of_cfs_init: OK
[    3.374911] Waiting for root device PARTUUID=874acb7a-02...
[    3.422360] mmc0: host does not support reading read-only switch, assuming write-enable
[    3.427800] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    3.430824] Indeed it is in host mode hprt0 = 00001101
[    3.495376] mmc0: new high speed SDHC card at address 59b4
[    3.499293] mmcblk0: mmc0:59b4 USDU1 14.9 GiB
[    3.505275]  mmcblk0: p1 p2
[    3.531340] mmc1: new high speed SDIO card at address 0001
[    3.538298] EXT4-fs (mmcblk0p2): mounted filesystem with ordered data mode. Opts: (null)
[    3.544455] VFS: Mounted root (ext4 filesystem) readonly on device 179:2.
[    3.555642] devtmpfs: mounted
[    3.566175] Freeing unused kernel memory: 1024K
[    3.584665] Run /sbin/init as init process
[    3.587590]   with arguments:
[    3.587602]     /sbin/init
[    3.587613]   with environment:
[    3.587624]     HOME=/
[    3.587635]     TERM=linux
[    3.664543] usb 1-1: New USB device found, idVendor=0424, idProduct=2514, bcdDevice= b.b3
[    3.670505] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    3.674599] hub 1-1:1.0: USB hub found
[    3.677825] hub 1-1:1.0: 4 ports detected
[    3.994193] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    4.124581] usb 1-1.1: New USB device found, idVendor=0424, idProduct=2514, bcdDevice= b.b3
[    4.131262] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.135752] hub 1-1.1:1.0: USB hub found
[    4.139380] hub 1-1.1:1.0: 3 ports detected
[    4.209962] systemd[1]: System time before build time, advancing clock.
[    4.358368] NET: Registered protocol family 10
[    4.363378] Segment Routing with IPv6
[    4.454186] usb 1-1.1.3: new full-speed USB device number 4 using dwc_otg
[    4.458468] systemd[1]: systemd 241 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid)
[    4.470886] systemd[1]: Detected architecture arm.
[    4.548538] systemd[1]: Set hostname to <octopi>.
[    4.609327] usb 1-1.1.3: New USB device found, idVendor=2341, idProduct=003d, bcdDevice= 0.01
[    4.617457] usb 1-1.1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=220
[    4.625515] usb 1-1.1.3: Product: Arduino Due Prog. Port
[    4.629457] usb 1-1.1.3: Manufacturer: Arduino (www.arduino.cc)
[    4.633464] usb 1-1.1.3: SerialNumber: 55839313638351F041C2
[    4.964189] usb 1-1.1.1: new high-speed USB device number 5 using dwc_otg
[    5.114817] usb 1-1.1.1: New USB device found, idVendor=0424, idProduct=7800, bcdDevice= 3.00
[    5.122048] usb 1-1.1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    5.394725] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): No External EEPROM. Setting MAC Speed
[    5.404095] libphy: lan78xx-mdiobus: probed
[    5.429292] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): int urb period 64
[    5.586117] random: systemd: uninitialized urandom read (16 bytes read)
[    5.605082] random: systemd: uninitialized urandom read (16 bytes read)
[    5.609668] systemd[1]: Started Dispatch Password Requests to Console Directory Watch.
[    5.621955] random: systemd: uninitialized urandom read (16 bytes read)
[    5.628007] systemd[1]: Created slice User and Session Slice.
[    5.637395] systemd[1]: Listening on Journal Socket (/dev/log).
[    5.646678] systemd[1]: Listening on udev Kernel Socket.
[    5.657495] systemd[1]: Created slice system-systemd\x2dfsck.slice.
[    5.667425] systemd[1]: Created slice system-getty.slice.
[    5.675948] systemd[1]: Reached target Slices.
[    6.471890] EXT4-fs (mmcblk0p2): re-mounted. Opts: (null)
[    6.600709] systemd-journald[120]: Received request to flush runtime journal from PID 1
[    7.202571] vc_sm_cma: module is from the staging directory, the quality is unknown, you have been warned.
[    7.206218] bcm2835_vc_sm_cma_probe: Videocore shared memory driver
[    7.206247] [vc_sm_connected_init]: start
[    7.215760] [vc_sm_connected_init]: installed successfully
[    7.232594] mc: Linux media interface: v0.10
[    7.294767] videodev: Linux video capture interface: v2.00
[    7.302484] snd_bcm2835: module is from the staging directory, the quality is unknown, you have been warned.
[    7.309878] bcm2835_audio bcm2835_audio: card created with 4 channels
[    7.316489] bcm2835_audio bcm2835_audio: card created with 4 channels
[    7.343180] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    7.345786] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    7.352918] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    7.381110] bcm2835_isp: module is from the staging directory, the quality is unknown, you have been warned.
[    7.383559] bcm2835_v4l2: module is from the staging directory, the quality is unknown, you have been warned.
[    7.391409] bcm2835-isp bcm2835-isp: Device node output[0] registered as /dev/video13
[    7.392202] bcm2835-isp bcm2835-isp: Device node capture[0] registered as /dev/video14
[    7.393698] bcm2835-isp bcm2835-isp: Device node capture[1] registered as /dev/video15
[    7.394187] bcm2835-isp bcm2835-isp: Device node stats[2] registered as /dev/video16
[    7.394230] bcm2835-isp bcm2835-isp: Register output node 0 with media controller
[    7.394257] bcm2835-isp bcm2835-isp: Register capture node 1 with media controller
[    7.394291] bcm2835-isp bcm2835-isp: Register capture node 2 with media controller
[    7.394313] bcm2835-isp bcm2835-isp: Register capture node 3 with media controller
[    7.394770] bcm2835-isp bcm2835-isp: Loaded V4L2 bcm2835-isp
[    7.400014] bcm2835_codec: module is from the staging directory, the quality is unknown, you have been warned.
[    7.733552] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    7.806247] cdc_acm 1-1.1.3:1.0: ttyACM0: USB ACM device
[    7.807477] usbcore: registered new interface driver cdc_acm
[    7.807498] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
[    7.859316] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    7.965602] brcmfmac: F1 signature read @0x18000000=0x15264345
[    7.984066] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    7.986113] usbcore: registered new interface driver brcmfmac
[    8.023230] bcm2835-codec bcm2835-codec: Device registered as /dev/video10
[    8.023291] bcm2835-codec bcm2835-codec: Loaded V4L2 decode
[    8.040019] bcm2835-codec bcm2835-codec: Device registered as /dev/video11
[    8.040250] bcm2835-codec bcm2835-codec: Loaded V4L2 encode
[    8.060153] bcm2835-codec bcm2835-codec: Device registered as /dev/video12
[    8.060227] bcm2835-codec bcm2835-codec: Loaded V4L2 isp
[    8.063753] bcm2835-codec bcm2835-codec: Device registered as /dev/video18
[    8.063806] bcm2835-codec bcm2835-codec: Loaded V4L2 image_fx
[    8.281087] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    8.281258] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    8.338096] brcmfmac: brcmf_c_preinit_dcmds: Firmware: BCM4345/6 wl0: Jan  4 2021 19:56:29 version 7.45.229 (617f1f5 CY) FWID 01-2dbd9d2e
[   10.027368] random: crng init done
[   10.027385] random: 7 urandom warning(s) missed due to ratelimiting
[   10.354844] uart-pl011 3f201000.serial: no DMA platform data
[   10.461774] 8021q: 802.1Q VLAN Support v1.8
[   10.464279] Adding 102396k swap on /var/swap.  Priority:-2 extents:1 across:102396k SSFS
[   10.994609] brcmfmac: brcmf_cfg80211_set_power_mgmt: power save enabled
[   11.367583] 8021q: adding VLAN 0 to HW filter on device eth0
[   16.282832] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   17.700042] Bluetooth: Core ver 2.22
[   17.700165] NET: Registered protocol family 31
[   17.700175] Bluetooth: HCI device and connection manager initialized
[   17.700206] Bluetooth: HCI socket layer initialized
[   17.700221] Bluetooth: L2CAP socket layer initialized
[   17.700250] Bluetooth: SCO socket layer initialized
[   17.712401] Bluetooth: HCI UART driver ver 2.3
[   17.712425] Bluetooth: HCI UART protocol H4 registered
[   17.712466] ICMPv6: process `dhcpcd' is using deprecated sysctl (syscall) net.ipv6.neigh.wlan0.retrans_time - use net.ipv6.neigh.wlan0.retrans_time_ms instead
[   17.712533] Bluetooth: HCI UART protocol Three-wire (H5) registered
[   17.712763] Bluetooth: HCI UART protocol Broadcom registered
[   18.057254] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   18.057269] Bluetooth: BNEP filters: protocol multicast
[   18.057296] Bluetooth: BNEP socket layer initialized
```

**Comment by Sineos** [comment]
I never used a Due with Klipper and I'm surely not an expert at it, but IIRC correctly the Due has two USB ports

`[    4.625515] usb 1-1.1.3: Product: Arduino Due Prog. Port`

sounds like the programming port. Does the other USB port make a difference?

**Comment by StefanMeGit** [comment]
Thanks, i found something out.

It looks like there was a big change in the past. I was able to connect to the MCU via the other port. 

I used the Prog. Port for 3 years. After the update i had to use the SamX dirct chip port to connect.... very strange. I also had to reconfigure all the PINS, because they have now different names.

Now i run into another issue.

For some reason my Prog. Port changes from time to time.

i got it working at the time the Pi recognice the port as:
```
pi@octopi:~ $ ls /dev/serial/by-id/*
serial: /dev/serial/by-id/usb-Klipper_sam3x8e_20312051513348323136303039303132-if00
```

but after a restart the usb name changed to:
```
pi@octopi:~ $ ls /dev/serial/by-id/*
serial: /dev/serial/by-id/usb-03eb_6124-if00
```
i tried to connect now via `serial: /dev/serial/by-id/usb-03eb_6124-if00` which doesnt work again. I need a frew restarts or some other random stuff to get the old name like serial: `/dev/serial/by-id/usb-Klipper_sam3x8e_20312051513348323136303039303132-if00` back.

very strange, Does anyone have a hint whats going on here?

thanks a lot
